src/DynamoPublish/DynamoPublishExtension.cs

```diff
@@ -54,5 +65,31 @@ private void OnMessageLogged(ILogMessage msg)
         }
 
         #endregion
+
+        #region Helpers
+
+        private MenuItem GenerateMenuItem()
+        {
+            MenuItem item = new MenuItem();
+            item.Header = Resource.DynamoViewMenuItemPublishTitle;
+
+            item.Click += (sender, args) =>
+                {
+                    if (publishWindow == null)
+                    {
+                        publishWindow = new PublishView(publishVM);
+                        publishWindow.Closed += (s, arg) =>
+                        {
+                            publishWindow = null;
+                        };
+                    }
+
+                    publishWindow.ShowDialog();
```

Comment:
Let's fully create the `PublishViewModel`, `PublishModel`, and `PublishView` here.  Do not hold onto any references.  as member fields.